### PR TITLE
fix queue stats to included 'processed' tasks

### PR DESCRIPTION
**Pull request by elv-gilles**
* add state for processed tasks
* add function in inspector to delete all processed tasks
* add synchronization on queues
* fix go mod in tools sub package

In the redis implementation processed keys are removed from the tasks set and a counter is incremented under a key of the form `asynq:{queue_name}processed:2006-01-02`. The same for failed tasks.

The current incarnation for [r/s]qlite instead keeps all tasks in the tasks table.
We may change it and implement something closer to the redis one, but - for now, as a matter of simplicity - I just fixed the queue stats and added a function `DeleteAllProcessedTasks` to the inspector to allow cleaning up.

### CODE REVIEW

File: queues.go [hunk 1]
```diff
@@ -143,6 +154,14 @@ func (p *QueuesConfig) Names() []string {
 	return uniq(names, len(p.queueConfig))
 }
 
+func (p *QueuesConfig) UpdateQueues(queues map[string]int) error {
```
Comment (lukseven) [inline queues.go:157]: I don't see this function called anywhere.

**Review by lukseven — APPROVED** [review]
As discussed previously, I think keeping stats is not a core function of Asynq. Instead, we have to make sure that we get rid of garbage. So we should either remove processed tasks as soon as they enter that state, or have a background cleanup mechanism that calls DeleteAllProcessedTasks periodically. I would personally opt for the former. Can be implemented in a follow-up PR.

**Comment by jslching** [comment]
Thanks, what is the functional difference between `TaskStateCompleted` and `TaskStateProcessed`?

**Comment by elv-gilles** [comment]
> Thanks, what is the functional difference between `TaskStateCompleted` and `TaskStateProcessed`?

None. Initially there were only `TaskStateProcessed` (and they were removed when done). Now there's also `TaskStateCompleted` by which tasks are subject to retention.

**Comment by elv-gilles** [comment]
> remove processed tasks as soon as they enter that state

I'll make another PR to remove them.

**Pull request merged**